examples/basic/main.tf

```diff
@@ -22,14 +26,24 @@ resource "upcloud_network" "network" {
 
 resource "upcloud_server" "webservers" {
   count    = 3
-  hostname = "webserver${count.index}"
-  title    = "webserver_${count.index}"
+  hostname = "${local.prefix}server-${count.index}"
+  title    = "${local.prefix}server-${count.index}"
   zone     = "pl-waw1"
-  plan     = "1xCPU-2GB"
+  plan     = "1xCPU-1GB"
+
+  # Use user-data script to install and launch http server on start
+  metadata  = true
+  user_data = file("${path.module}/user-data-script.sh")
 
   template {
     storage = "Ubuntu Server 20.04 LTS (Focal Fossa)"
     size    = 25
+    title   = "${local.prefix}storage-${count.index}"
+  }
+
+  # Required to have internet access when installing httpd on launch
```

Comment:
Alternative would be to use similar script than in [tests](https://github.com/UpCloudLtd/terraform-upcloud-basic-loadbalancer/blob/main/test/basic/main.tf#L54) that does not require installations.

Reply:
I think that using separate script file is more clear and resulting plan is same either way.